src/JobHelper.groovy

```diff
@@ -62,23 +62,53 @@ class JobHelper {
     /**
     * Queries an api and returns the results as a JSON object
     * @param query
+    * @param attempts
+    * @param context
     * @return
     */
-    private static queryApi(String query) {
-        try {
-            def get = new URL(query).openConnection()
-            get.setRequestProperty("User-Agent", "adopt-jenkins-helper")
-            def parser = new JsonSlurper()
-            return parser.parseText(get.getInputStream().getText())
-        } catch (Exception e) {
-            throw new RuntimeException("${e.getMessage()}")
+    private static queryApi(String query, Integer attempts, def context) {
+        def jsonResponse = null
+        def parser = new JsonSlurper()
+
+        for (int count = 1; count <= attempts; count++) {
+            try {
+                def get = new URL(query).openConnection()
+                get.setRequestProperty("User-Agent", "adopt-jenkins-helper")
+                jsonResponse = parser.parseText(get.getInputStream().getText())
+
+                // Successful response
+                if (jsonResponse != null) {
+                    context.println "[SUCCESS] We have a response!\n${jsonResponse}"
+                    break
+                } else {
+                    context.println "[RETRYWARNING] API Request was successful but jsonResponse is null. Retrying in 30 seconds..."
+                    context.sleep 30000
+                }
+            } catch (Exception e) {
+                if (count >= attempts) {
```

Comment:
how would count > attempts?

Reply:
I used >= just as insurance. It should only need to be ==

Reply:
Right, worth changing that not to confuse the next reader.

Reply:
Done